frame/vamm/src/tests/update_twap.rs

```diff
@@ -147,22 +184,23 @@ fn update_twap_succeeds() {
 		peg_multiplier: 1,
 		..Default::default()
 	};
+	let new_twap_reciprocal = Some(TestPallet::reciprocal_twap(&new_twap.unwrap()).unwrap());
 	ExtBuilder { vamm_count: 1, vamms: vec![(0, vamm_state)] }
 		.build()
 		.execute_with(|| {
-			run_for_seconds(1);
+			run_for_seconds(twap_update_delay(0));
 			assert_ok!(
-				TestPallet::update_twap(0, new_twap, new_twap),
-				(new_twap.unwrap(), new_twap.unwrap())
+				TestPallet::update_twap(0, new_twap, new_twap_reciprocal),
+				(new_twap.unwrap(), new_twap_reciprocal.unwrap())
 			);
 
-			run_for_seconds(1);
+			run_for_seconds(twap_update_delay(0));
 			assert_ok!(TestPallet::update_twap(0, None, new_twap));
 
-			run_for_seconds(1);
+			run_for_seconds(twap_update_delay(0));
 			assert_ok!(TestPallet::update_twap(0, new_twap, None));
 
-			run_for_seconds(1);
+			run_for_seconds(twap_update_delay(0));
```

Comment:
It seems that these calls are advancing time to
```now + twap_update_delay = now + (vamm_state.twap_timestamp + vamm_state.twap_period + 1)```
since `run_for_seconds` increments the **current** timestamp by its argument. 

Are you sure you didn't mean to do only the following?
```now + vamm_state.twap_period + 1```
That would be `run_for_seconds(vamm_state.twap_period + 1)`

Or you could use a different function like `run_to_seconds(...)` that advances time **to** its argument, which I think is what you intended here.

Reply:
Oh, thanks for pointing out!
Yeah I was using it like I use `run_to_block`. 
Fixed it :+1: